src/frontends/pytorch/src/op/atan2.cpp

```diff
@@ -0,0 +1,100 @@
+    auto const_pi = v0::Constant::create(element::f32, Shape{}, {std::atan(1.0)*4});
+    align_eltwise_input_types(context, x, const_pi, false, true);
+    auto is_x_negative = context.mark_node(std::make_shared<v1::Less>(x, const_zero));
+    align_eltwise_input_types(context, y, const_zero, false, true);
+    auto y_non_negative = context.mark_node(std::make_shared<v1::GreaterEqual>(y, const_zero));
+    auto cond1 = context.mark_node(std::make_shared<v1::LogicalAnd>(is_x_negative, y_non_negative));
+    align_eltwise_input_types(context, atan, const_pi, false, true);
+    auto atan_y_x_plus_pi = context.mark_node(std::make_shared<v1::Add>(atan, const_pi));
+    result = context.mark_node(std::make_shared<v1::Select>(cond1, atan_y_x_plus_pi, result));
+
+    // handle the third condition : x<0 && y<0
+    align_eltwise_input_types(context, x, const_zero, false, true);
+    auto is_y_negative = context.mark_node(std::make_shared<v1::Less>(y, const_zero));
+    auto cond2 = context.mark_node(std::make_shared<v1::LogicalAnd>(is_x_negative, is_y_negative));
+    align_eltwise_input_types(context, atan, const_pi, false, true);
+    auto atan_y_x_minus_pi = context.mark_node(std::make_shared<v1::Subtract>(atan, const_pi));
+    result = context.mark_node(std::make_shared<v1::Select>(cond2, atan_y_x_minus_pi, result));
+
+    // handle the fourth condition : x=0 && y>0
+    align_eltwise_input_types(context, x, const_zero, false, true);
+    auto is_x_zero = context.mark_node(std::make_shared<v1::Equal>(x, const_zero));
+    align_eltwise_input_types(context, y, const_zero, false, true);
+    auto is_y_positive = context.mark_node(std::make_shared<v1::Greater>(y, const_zero));
+    auto cond3 = context.mark_node(std::make_shared<v1::LogicalAnd>(is_x_zero, is_y_positive));
+    auto const_two = v0::Constant::create(element::f32, Shape{}, {2});
+    align_eltwise_input_types(context, const_pi, const_two, false, true);
+    auto pi_div_two = context.mark_node(std::make_shared<v1::Divide>(const_pi, const_two));
+    result = context.mark_node(std::make_shared<v1::Select>(cond3, pi_div_two, result));
+
+    // handle the fifth condition : x=0 && y<0
+    auto cond4 = context.mark_node(std::make_shared<v1::LogicalAnd>(is_x_zero, is_y_negative));
+    auto const_minus_two = v0::Constant::create(element::f32, Shape{}, {-2});
+    align_eltwise_input_types(context, const_pi, const_minus_two, false, true);
+    auto pi_div_minus_two = context.mark_node(std::make_shared<v1::Divide>(const_pi, const_minus_two));
+    result = context.mark_node(std::make_shared<v1::Select>(cond4, pi_div_two, result));
+
+    // check whether out tensor is given
+    if(!context.input_is_none(2) && context.get_input_size == 3){
+    out_tensor = context.get_input(2);
+    // dtype is inherited from out tensor in input
+    auto result_out = context.mark_node(std::make_shared<v1::ConvertLike>(result,out_tensor);
+
+    return {result_out};
```

Comment:
In this code, out_tensor was not declared (missing auto) and in ConvertLike line parentheses are not closed causing build to fail. To handle out tensor, it would also need to use context.mutate_input(...).
If out is always on input with index 2, maybe it would be better to use `op::optional_out` helper in op_table.cpp?
You would need to modify `{"aten::atan2", op::translate_atan2},` to `{"aten::atan2", op::optional_out<op::translate_atan2, 2>},`. You can look at `aten::rsqrt` as an example.

Reply:
Hi! @mmikolajcz thanks, sure will take a look